content/blog/accessing-hpe-data-fabric-s3-storage-from-spring-boot-s3-micro-service-deployed-in-k3s-cluster.md

```diff
@@ -0,0 +1,155 @@
+
+**Data Fabric Object Storage**
+
+The Object Store with an S3-compatible API stores data generated through multiple data protocols, such as NFS, POSIX, S3, and HDFS. Data in the Object Store is accessible through S3 API requests. The Object Store manages all inbound S3 API requests to store data in or retrieve data from an HPE Ezmeral Data Fabric cluster. More details can be found [here.](https://docs.datafabric.hpe.com/62/MapRObjectStore/MapRObjectStorewithS3-compatibleAPI.html)
+
+**K3s**
+
+Lightweight Kubernetes, aka K3s, is easy to install and consumes half the memory, all in a binary of less than 100MB. It's great for edge and IoT use cases. More information on K3s can be found at [Rancher](<https://rancher.com/docs/k3s/latest/en/>) site[](https://rancher.com/docs/k3s/latest/en/) . Follow the steps as mentioned in [QuickStart Guide](https://rancher.com/docs/k3s/latest/en/quick-start/) [](https://rancher.com/docs/k3s/latest/en/quick-start/) for installation of the K3s cluster. 
+
+**Spring Boot**
+
+Spring Boot is an open source Java-based framework used to create a micro Service. Many Real world applications are written in Spring Boot for faster development and better maintainability. More information can be found at** [spring.io](https://spring.io/projects/spring-boot)**
+
+Note: We can move the Spring Boot application to Quarkus with no change in code.  Moving code to Quarkus will reduce the footprint of Spring Boot application. More information can be found at [Quarkus site](https://quarkus.io/blog/quarkus-for-spring-developers/)
+
+In this blog, checkout existing Spring Boot application from [GitHub](https://github.hpe.com/kiran-mavatoor/df-s3-springboot-k3s-demo) , customise it and execute. 
+
+**Application Prerequisites**
+
+1.      K3s cluster must be accessible. Note down the control plan node details. This information is required to deploy the Spring Boot application.
+
+2.      Access the HPE Data Fabric Object Store service UI running on port 9000. For example URL  - https://<Fully Qualified Domain Name>:9000/ .  Note down the access key and secret key. It is advised to change the default values.
+
+3.      Java 11 , Apache Maven 3.8+ , Docker Client.
+
+**Build and Install Steps**
+
+1.      Check out  an existing Spring Boot application from [github](<https://github.hpe.com/kiran-mavatoor/df-s3-springboot-k3s-demo>).
+
+2.      Copy the ssl_usertruststore.p12 from the HPE Data Fabric cluster into certs folder under project directory. The ssl_usertruststore.p12 file is located at /opt/mapr/conf directory in cluster node. The password for p12 can be copied from “ssl.client.truststore.password” property value in /opt/mapr/conf/ssl-client.xml .
+
+3.      From project directory, open resources/application.properties. Change the key values as per your environment. 
+
+4.      Execute “mvn clean install” .
+
+5.      The distributable is available in target/df-s3-springboot-k3s-demo-1.0-SNAPSHOT.jar .
```

Comment:
I see what you're doing here by adding the space prior to the period at the end of the sentence. However, for Step 5 it pushes it down to the next line. You may just want to eliminate the periods entirely with the bullets so as to avoid this issue.

![image](https://user-images.githubusercontent.com/77496261/134180510-81406422-af1d-45af-8aef-4949d8c8eeab.png)